test/strings/basic.jl

```diff
@@ -176,6 +176,38 @@ gstr = GenericString("12");
 @test getindex(gstr,Bool(1):Bool(1))=="1"
 @test getindex(gstr,AbstractVector([Bool(1):Bool(1);]))=="1"
 
+# This gets a deprecation warning that seems incorrect
+# getindex("a",1.0) -> WARNING: indexing with non Integer Reals is deprecated
+# Capture spurious deprecation warning
+let
```

Comment:
Should the getindex call itself be moved to `deprecated.jl` then, since it depends on a deprecated method (`to_index(x::Real)`)?
I was just trying to make sure that every line in strings was covered.